### PR TITLE
Use a service account to access other API's

### PR DESCRIPTION
**Pull request by Matt-Guest**
### What

Added service account token to HTTP requests going out. 

On startup the service account token is check, to see if its valid. If this check is
failed the app shall exit.

### How to review

Review the code, run the unit tests and run some manual tests

### Who can review

anyone

### CODE REVIEW

File: auth/auth.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
+package auth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/ONSdigital/go-ns/rchttp"
+	"github.com/ian-kent/go-log/log"
```
Comment (mattrout92) [inline auth/auth.go]: This needs updating to be the go-ns log package

File: auth/auth.go [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+)
+
+func CheckServiceIdentity(ctx context.Context, zebedeeURL, serviceAuthToken string) error {
+	client := rchttp.DefaultClient
+
+	path := fmt.Sprintf("%s/identity", zebedeeURL)
+
+	var URL *url.URL
```
Comment (mattrout92) [inline auth/auth.go:19]: url should not be exported

**Review by mattrout92 — DISMISSED** [review]

File: filter/store.go [hunk 3]
```diff
@@ -21,7 +21,8 @@ import (
 // Store provides access to stored dimension data.
 type Store struct {
 	filterAPIURL       string
-	filterAPIAuthToken string
+	filterAPIAuthToken string // Deprecated variable
+	serviceToken       string
```
Comment (mattrout92) [inline filter/store.go:25]: This is not being set anywhere

File: auth/auth.go (hunk 1 above)
Comment (CarlHembrough) [reply auth/auth.go]: Done 👍

File: auth/auth.go (hunk 2 above)
Comment (CarlHembrough) [reply auth/auth.go:19]: It won't be exported as its declared in a function?

File: filter/store.go (hunk 3 above)
Comment (CarlHembrough) [reply filter/store.go:25]: Added it into the constructor function

**Review by nshumoogum — APPROVED** [review]
👍

File: README.md [hunk 4]
```diff
@@ -30,6 +32,8 @@ environment variables, or with a link to a configuration guide.
 | ERROR_PRODUCER_TOPIC        | filter-error                         | The topic to add messages to when an error occurs
 | HEALTHCHECK_INTERVAL        | time.Minute                          | How often to run a health check
 | GRACEFUL_SHUTDOWN_TIMEOUT   | time.Second * 10                     | How long to wait for the service to shutdown gracefully
+| SERVICE_AUTH_TOKEN          | 939616dc-7599-4ded-9a86-a9c66fbf98e0 | The service token for this app
```
Comment (nshumoogum) [inline README.md]: Update default value to `0f49d57b-c551-4d33-af1e-a442801dd851`

File: config/config.go [hunk 5]
```diff
@@ -47,6 +49,8 @@ func Get() (*Config, error) {
 		ErrorProducerTopic:       "filter-error",
 		GracefulShutdownTimeout:  time.Second * 10,
 		HealthCheckInterval:      time.Minute,
+		ServiceAuthToken:         "939616dc-7599-4ded-9a86-a9c66fbf98e0",
```
Comment (nshumoogum) [inline config/config.go]: Update default value to `0f49d57b-c551-4d33-af1e-a442801dd851`

Comment (nshumoogum) [reply config/config.go]: And then append value with `Bearer ` after line 54

`cfg.ServiceAuthToken = "Bearer " + cfg.ServiceAuthToken`

**Review by nshumoogum — CHANGES_REQUESTED** [review]

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**